packages/block-library/src/navigation/index.php

```diff
@@ -121,7 +132,8 @@ function build_navigation_html( $block, $colors, $font_sizes ) {
 		? sprintf( ' style="%s"', esc_attr( $colors['inline_styles'] ) . esc_attr( $font_sizes['inline_styles'] ) )
 		: '';
 
-	foreach ( (array) $block['innerBlocks'] as $key => $block ) {
+	// Only include link blocks that have a label set.
+	foreach ( (array) array_filter( $block['innerBlocks'], 'has_navigation_label' ) as $key => $block ) {
```

Comment:
A suggestion in order to run this write this call once:

```php
function render_block_navigation( $attributes, $content, $block ) {
  $inner_blocks = (array) array_filter( $block['innerBlocks'], 'has_navigation_label' );
  if ( empty( $inner_blocks ) ) {
    return ;
  }

  // ...
}
```

In this way, we are avoiding run unnecessary code such as defining vars, etc. And also we won't need to care for checking the sub-menus items (L181)

Reply:
If we're only using the callback once, it's fine to keep it inline, too.

Reply:
Do you mean that the `return;` will break the loop?